light-client/src/tests.rs

```diff
@@ -66,3 +66,77 @@ impl Trusted {
         }
     }
 }
+
+// -----------------------------------------------------------------------------
+// Everything below is a temporary workaround for the lack of `provider` field
+// in the light blocks serialized in the JSON fixtures.
+// -----------------------------------------------------------------------------
```

Comment:
cc @greg-szabo @Shivani912  this is relevant to testing / serialization

Reply:
This seems to be implementation specific to me. Go code has a different approach to identify a node as a primary/secondary provider and does not deal with addresses. But also, this approach looks cleaner and reduces a lot of data from test files. Will need some more work around testing of Go code though. Need more thoughts here! @greg-szabo @Liamsi @melekes